### ISSUE TITLE
WiFi does not reconnect after long sleep

### ISSUE THREAD

**Issue by UniversalSuperBox** [ubports/ubuntu-touch#1604]
- Device: Nexus 6P (angler)
- Channel: devel
- Build: 2020-10-08

### Steps to reproduce

1. Connect to a WiFi network
1. Lock your device
1. Wait 8-10 hours
1. Wake and unlock your device

### Expected behavior

WiFi is still connected

### Actual behavior

WiFi is disconnected.

### Logfiles and additional information

None provided

**Comment by Flohack74** [comment]
I can confirm this. Its basically the WiFi driver racing with power management: It wants to wake up WiFi, but the PCIe bus is still not available. Then the driver tears down the whole card and goes dormant. I did not find anything on the internet for it so far.

**Comment by Boballen55** [comment]
This has been the most consistantly bithersome bug on the 6p. Just poking this issue to let you know there is someone out there that cares about this issue and appreciates all your efforts.

**Comment by Flohack74** [comment]
Yeah :) and its not only on this device. Must be kinda race condition, also, why does the WiFi disconnect at all in sleep, it should not. Maybe its your AP that drops a device after its idle too long? Then, push client will generate traffic every 5mins...
